test/XRP/protos-models-flags-utils/gateway-balances.test.ts

```diff
@@ -0,0 +1,257 @@
+import { assert } from 'chai'
+
+import GatewayBalances, {
+  gatewayBalancesFromResponse,
+} from '../../../src/XRP/shared/gateway-balances'
+import 'mocha'
+import { GatewayBalancesResponse } from '../../../src/XRP/shared/rippled-json-rpc-schema'
+import { XrpUtils } from 'xpring-common-js'
+
+describe('GatewayBalances Conversion Tests', function (): void {
+  it('GatewayBalances from GatewayBalancesResponse - all fields present', function (): void {
+    // GIVEN a raw JSON GatewayBalancesResponse
+    const testGatewayBalancesResponse: GatewayBalancesResponse = {
+      result: {
+        account: 'rMwjYedjc7qqtKYVLiAccJSmCwih4LnE2q',
+        assets: {
+          r9F6wk8HkXrgYWoJ7fsv4VrUBVoqDVtzkH: [
+            {
+              currency: 'BTC',
+              value: '5444166510000000e-26',
+            },
+            {
+              currency: 'EUR',
+              value: '7000000000000000e-26',
+            },
```

Comment:
this fake looks pretty similar to the one in the next test - can we make a base object that has the required fields and compose more complicated fakes? it's hard to tell at a glance what's missing and what's not

Reply:
that's a great idea